test/CommandLineUtils.Tests/DefaultHelpTextGeneratorTests.cs

```diff
@@ -85,5 +85,39 @@ private string GetHelpText(CommandLineApplication app)
             _output.WriteLine(helpText);
             return helpText;
         }
+        
+        enum SomeEnum { None, Normal, Extreme }
+
+        [Fact]
+        public void ShowHelp()
+        {
+            var app = new CommandLineApplication();
+            app.HelpOption();
+            app.Option("--strOpt <E>", "int option desc", CommandOptionType.SingleValue);
+            app.Option<int>("--intOpt <E>", "int option desc", CommandOptionType.SingleValue);
+            app.Option<SomeEnum>("--enumOpt <E>", "enum option desc", CommandOptionType.SingleValue);
+            app.Argument("SomeArgument1", "arg desc");
+            app.Argument<SomeEnum>("SomeArgument2", "arg desc");
+
+            StringWriter sw = new StringWriter();
+            app.Out = sw;
+            app.ShowHelp(false);
+
+            Assert.Equal(@"Usage:  [options] <SomeArgument1> <SomeArgument2>
+
+Arguments:
+  SomeArgument1  arg desc
+  SomeArgument2  arg desc
+                 Allowed values are: None, Normal, Extreme
+
+Options:
+  -?|-h|--help   Show help information
+  --strOpt <E>   int option desc
+  --intOpt <E>   int option desc
+  --enumOpt <E>  enum option desc
+                 Allowed values are: None, Normal, Extreme
+
+", sw.ToString());
```

Comment:
```suggestion
",
            sw.ToString(),
            ignoreLineEndingDifferences: true);
```

Multi-line string literals + git + assertions is a recipe for trouble. Ignoring line endings makes this more reliable.